src/main/java/org/myrobotlab/service/KeyboardSim.java

```diff
@@ -0,0 +1,157 @@
+package org.myrobotlab.service;
+
+import java.awt.AWTException;
+import java.awt.Robot;
+import java.awt.event.KeyEvent;
+
+import org.myrobotlab.framework.Service;
+import org.myrobotlab.framework.ServiceType;
+import org.myrobotlab.logging.Level;
+import org.myrobotlab.logging.LoggerFactory;
+import org.myrobotlab.logging.LoggingFactory;
+import org.slf4j.Logger;
+
+/**
+ * @author MaVo (MyRobotLab) / LunDev (GitHub)
+ */
+
+public class KeyboardSim extends Service {
+	static final long serialVersionUID = 1L;
+	static final Logger log = LoggerFactory.getLogger(KeyboardSim.class);
+
+	private Robot robot;
+
+	public KeyboardSim(String n) {
+		super(n);
+	}
+
+	@Override
+	public void startService() {
+		super.startService();
+		try {
+			robot = new Robot();
+		} catch (AWTException e) {
+			log.error("could not create java.awt.Robot");
+			log.error(e.getMessage());
+		}
+	}
+
+	/**
+	 * simulate pressing and releasing a keyboard key after the desired duration
+	 * 
+	 * @param keycode  keycode of key (use KeyEvent.VK_XXX, where XXX is your key)
+	 * @param duration key press duration in milliseconds
+	 */
+	public void pressAndRelease(int keycode, int duration) {
+		press(keycode);
+		try {
+			Thread.sleep(duration);
+		} catch (InterruptedException e) {
+		}
+		release(keycode);
+	}
+
+	/**
+	 * simulate pressing a keyboard key (does NOT release the key!)
+	 * 
+	 * @param keycode  keycode of key (use KeyEvent.VK_XXX, where XXX is your key)
+	 * @param duration key press duration in milliseconds
+	 */
+	public void press(int keycode) {
+		robot.keyPress(keycode);
+	}
+
```

Comment:
Need String functions .. so happy users can 
keyboard.press("A");   !!!!

```java
public void press(String key)
```

Reply:
I will add this, but I think this will result in a huge switch block : )
Will still look for a better solution.

Reply:
How about this ?
https://stackoverflow.com/questions/15260282/converting-a-char-into-java-keyevent-keycode

Reply:
ya, something like this, might get funky when trying to support non alpha-numeric characters though - needs testing.